docs/data/data-grid/demo/PopularFeaturesDemo.tsx

```diff
@@ -0,0 +1,383 @@
+    maxWidth: 160,
+    flex: 0.2,
+    minWidth: 100,
+    groupable: false,
+    renderCell: (params) => {
+      if (!params.value) {
+        return <React.Fragment />;
+      }
+      return (
+        <Box
+          sx={{
+            width: '100%',
+            height: '100%',
+            position: 'relative',
+            display: 'flex',
+            alignItems: 'center',
+          }}
+        >
+          <Typography sx={{ fontSize: '1rem', fontWeight: '500' }}>
+            <Link
+              href={`/x/react-data-grid${params.row.detailPage}`}
+              target="_blank"
+            >
+              {params.value}
+            </Link>
+          </Typography>
+          {params.row.newBadge && (
+            <Box
+              sx={{
+                width: 'fit-content',
+                height: 'fit-content',
+                fontSize: '0.8em',
+                fontWeight: 600,
+                position: 'absolute',
+                textAlign: 'center',
+                top: -13,
+                left: -20,
+                background: '#fcf0a0',
+                color: '#af5b00',
+                px: 1,
+                borderRadius: 10,
+              }}
+            >
+              New
+            </Box>
+          )}
+        </Box>
+      );
+    },
+  },
+  {
+    field: 'description',
+    headerName: 'Brief description',
+    groupable: false,
+    flex: 0.5,
+    minWidth: 120,
+  },
+  {
+    field: 'plan',
+    headerName: 'Plan',
+    maxWidth: 130,
+    flex: 0.3,
+    type: 'singleSelect',
+    valueOptions: ['Premium', 'Pro', 'Community'],
+    renderCell: (params: GridRenderCellParams<string>) => {
+      if (!params.value) {
+        return <React.Fragment />;
+      }
+      return <PlanTag plan={params.value} />;
+    },
+    sortComparator: (p1, p2) => {
+      function getSortingValue(plan: string) {
+        switch (plan) {
+          case 'Pro':
+            return 1;
+          case 'Premium':
+            return 2;
+          default:
+            return 0;
+        }
+      }
+      const p1Value = getSortingValue(p1);
+      const p2Value = getSortingValue(p2);
+      return p1Value - p2Value;
+    },
+  },
+];
+
+export default function PopularFeaturesDemo() {
+  const getDetailPanelContent = React.useCallback(
+    ({ row }) => <RowDemo row={row} />,
+    [],
+  );
+
+  return (
+    <div style={{ height: 'fit-content', width: '100%' }}>
+      <DataGridPremium
+        autoHeight
+        disableSelectionOnClick
+        onCellClick={(_, event) => event.stopPropagation()}
+        components={{
+          Toolbar: CustomToolbar,
+          DetailPanelExpandIcon: ArrowDown,
+          DetailPanelCollapseIcon: ArrowUp,
+        }}
+        componentsProps={{
+          toolbar: { showQuickFilter: true, quickFilterProps: { debounceMs: 500 } },
+        }}
+        getDetailPanelContent={getDetailPanelContent}
+        getDetailPanelHeight={() => 'auto'}
+        getRowHeight={() => 'auto'}
```

Comment:
`getRowHeight` and `groupingColDef` should be memoized
I could not reproduce a bug because the demo has no external state, but in the doc we say to always memoize them so I would follow our guidelines here :+1:

Reply:
Should we memoize getRowHeight even when it returns 'auto'? we should then refine our demo to only showcase like that.

Reply:
I am not fully aware of the consequences of a non-memoized callback
On a real dataset it has performances issues for sure
But I don't know what bugs it can also cause

Reply:
cc @m4theushw